src/stream_from.cxx

```diff
@@ -77,48 +109,34 @@ pqxx::stream_from::~stream_from() noexcept
 }
 
 
-bool pqxx::stream_from::get_raw_line(std::string &line)
+pqxx::stream_from::raw_line pqxx::stream_from::get_raw_line()
 {
-  internal::gate::connection_stream_from gate{m_trans.conn()};
   if (*this)
   {
+    internal::gate::connection_stream_from gate{m_trans.conn()};
     try
     {
-      if (not gate.read_copy_line(line))
+      raw_line line{gate.read_copy_line()};
+      if (line.first.get() == nullptr)
         close();
+      return line;
     }
     catch (std::exception const &)
     {
       close();
       throw;
     }
   }
-  return *this;
-}
-
-
-void pqxx::stream_from::set_up(
-  transaction_base &tb, std::string_view table_name)
-{
-  set_up(tb, table_name, "");
-}
-
-
-void pqxx::stream_from::set_up(
-  transaction_base &tb, std::string_view table_name,
-  std::string const &columns)
-{
-  // Get the encoding before starting the COPY, otherwise reading the
-  // variable will interrupt it.
-  m_copy_encoding = internal::enc_group(m_trans.conn().encoding_id());
-  begin_copy(tb, table_name, columns);
-  register_me();
+  else
+  {
+    return raw_line{};
+  }
 }
 
 
 void pqxx::stream_from::close()
 {
-  if (!m_finished)
+  if (not m_finished)
```

Comment:
Woooah, What the heck is "not" ?

Reply:
https://en.cppreference.com/w/cpp/language/operator_alternative

On our code base (we have a lot of people who mostly do Python) we decided that we preferred the readability of `not` over `!`.

Reply:
That's abhorrent.

Best of luck with that.